plugin/my_asset_manager/AssetManagerInterface.py

```diff
@@ -0,0 +1,95 @@
+#
+#   Copyright 2013-20213[The Foundry Visionmongers Ltd]
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# pylint: disable=invalid-name
+"""
+A single-class module, providing the MyAssetManagerInterface class.
+This is the entry-point for the logic of your asset manager.
+"""
+
+# Note that it should always be light-weight to construct instances of
+# the this class. See the notes under the "Initialization" section of:
+# https://openassetio.github.io/OpenAssetIO/classopenassetio_1_1v1_1_1manager_api_1_1_manager_interface.html#details (pylint: disable=line-too-long)
+# As such, any expensive module imports should be deferred.
+from openassetio import constants, BatchElementError, TraitsData
+from openassetio.managerApi import ManagerInterface
+
+# OpenAssetIO is building out the implementation vertically, there are
+# known fails for missing abstract methods.
+# pylint: disable=abstract-method
+# Methods in C++ end up with "missing docstring"
+# pylint: disable=missing-docstring
+# pylint: disable=too-many-arguments, unused-argument
+
+
+class MyAssetManagerInterface(ManagerInterface):
+    """
+    Implement the OpenAssetIO ManagerInterface.
+    https://openassetio.github.io/OpenAssetIO/classopenassetio_1_1manager_api_1_1_manager_interface_1_1_manager_interface.html
+    """
+
+    # Entity references provided to this asset manager should be
+    # prefixed with this string to be considered valid.
+    # eg. "my_asset_manager:///my_entity_id"
+    __reference_prefix = "my_asset_manager:///"
+
+    def identifier(self):
+        return "myorg.manager.my_asset_manager"
+
+    def initialize(self, managerSettings, hostSession):
+        pass
```

Comment:
I think we discovered that this does the correct thing by default? Have added some doc explaining that this is the place to put your initialization logic if you have any.

Reply:
I think we worked out that its pure virtual. It would need to assert that settings are empty to be correct. I think we need to update the apiComplianceSuite to check it asserts if no valid settings are provided in the fixture.